File: src/main/java/no/nav/veilarboppfolging/domain/ArenaSyncOppfolgingsBruker.java
package no.nav.veilarboppfolging.domain;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import no.nav.common.types.identer.AktorId;
import no.nav.pto_schema.enums.arena.Formidlingsgruppe;
import no.nav.pto_schema.enums.arena.Kvalifiseringsgruppe;
import no.nav.veilarboppfolging.repository.entity.OppfolgingStartBegrunnelse;

@EqualsAndHashCode(callSuper = true)
public class ArenaSyncOppfolgingsBruker extends Oppfolgingsbruker {
    @Getter
    Kvalifiseringsgruppe kvalifiseringsgruppe;
    Formidlingsgruppe formidlingsgruppe;

    ArenaSyncOppfolgingsBruker(AktorId aktorId, Formidlingsgruppe formidlingsgruppe, Kvalifiseringsgruppe kvalifiseringsgruppe) {
        super(aktorId.get(), formidlingsgruppe == Formidlingsgruppe.IARBS ? OppfolgingStartBegrunnelse.ARENA_SYNC_IARBS : OppfolgingStartBegrunnelse.ARENA_SYNC_ARBS, StartetAvType.SYSTEM);
        this.kvalifiseringsgruppe = kvalifiseringsgruppe;
    }
}

File: src/main/kotlin/no/nav/veilarboppfolging/oppfolgingsbruker/ArenaSyncOppfolgingsBruker.kt
package no.nav.veilarboppfolging.oppfolgingsbruker

import lombok.EqualsAndHashCode
import no.nav.common.types.identer.AktorId
import no.nav.pto_schema.enums.arena.Formidlingsgruppe
import no.nav.pto_schema.enums.arena.Kvalifiseringsgruppe
import no.nav.veilarboppfolging.domain.StartetAvType

@EqualsAndHashCode(callSuper = true)
data class ArenaSyncOppfolgingsBruker(
    override val aktorId: AktorId,
    val formidlingsgruppe: Formidlingsgruppe,
    val kvalifiseringsgruppe: Kvalifiseringsgruppe,
) : Oppfolgingsbruker(
    aktorId,
    if (formidlingsgruppe == Formidlingsgruppe.IARBS) OppfolgingStartBegrunnelse.ARENA_SYNC_IARBS else OppfolgingStartBegrunnelse.ARENA_SYNC_ARBS,
    StartetAvType.SYSTEM
)
